### DISCUSSION TITLE
local_temperature_calibration should accept float values without round

### DISCUSSION THREAD

**Discussion by KartoffelToby** [Koenkk/zigbee2mqtt#9942] [Devices]
whats the main reason that some Thermostat devices keep and uses float values correct and others change it to int and round them? i think the behavior should always be the same.

**Comment by MJeka** [comment]
> whats the main reason that some Thermostat devices keep and uses float values correct and others change it to int and round them? i think the behavior should always be the same.

The calibration 1 degree is very large. If the device supports 0.5 degrees, then it should be 0.5. You can configure all this in automations

**Reply by KartoffelToby** [reply]
are there devices that supports only 1 degree when there are use 2 decimal floats for the rest? 

 failed (The value of "value" is out of range. It must be >= 0 and <= 255. Received 54_975_581_389) i set it to 0.8
